cita-executor/core/src/cita_executive.rs

```diff
@@ -33,23 +34,26 @@ const AMEND_KV_H256: u32 = 3;
 const AMEND_ACCOUNT_BALANCE: u32 = 5;
 
 // FIXME: CITAExecutive need rename to Executive after all works ready.
-pub struct CitaExecutive<B> {
+pub struct CitaExecutive<'a, B> {
     block_provider: Arc<BlockDataProvider>,
     state_provider: Arc<RefCell<State<B>>>,
+    native_factory: &'a NativeFactory,
     env_info: EnvInfo,
     economical_model: EconomicalModel,
 }
 
-impl<B: DB + 'static> CitaExecutive<B> {
+impl<'a, B: DB + 'static> CitaExecutive<'a, B> {
     pub fn new(
         block_provider: Arc<BlockDataProvider>,
         state_provider: State<B>,
+        native_factory: &'a NativeFactory,
```

Comment:
change to `native_provider`?

Reply:
If you carefully taste, `Factory` and `provider` are different, I think it is more reasonable to use `Factory` here.

Reply:
OK，not a problem.